### ISSUE TITLE
A weird observation in document layout(grounding) task, relating to the min/max pixels settings

### ISSUE THREAD

**Issue by puppyapple** [QwenLM/Qwen2.5-VL#1311]
(this is a follow-up of #1062 )
I'm currently fine-tuning qwen2.5 vl 7b model for document layout task(which is essentially a grounding task). 
> To be clear with the context, I pre-processed all my bbox coordinates in response value with the resized input width/height output by **qwen_vl_utils.smart_size** function, setting the min_pixels=512 * 28 * 28, max_pixels=2048 * 28 * 28; And certainly I run all the inferences with exactly the same settings.

And when reviewing my bad case I found an interesting and weird phenomenon.
One of my bad cases gives the display below(which is a mess):

![Image](https://github.com/user-attachments/assets/60228a85-0936-4a12-81ab-05b1b1c87e84)

the smart_resize function offers **input_width: 1064, input_height: 1484** as the real image input size to the model.

And then I made a slight change with the max_pixels, from **2048 * 28 * 28** to **2000 * 28 * 28** when preprocessing the input image, but which will cause the change of the input_width/input_height, from **input_width: 1064, input_height: 1484** below to **input_width: 1036, input_height: 1484**. Magically the output turns out to be:

![Image](https://github.com/user-attachments/assets/bd14da98-45b9-4cf7-b2a3-e10a0efcf91f)

which is a pretty good grounding result.

So I wonder how this interesting situation could happen?
@JJJYmmm

**Comment by JJJYmmm** [comment]
Currently 2.5VL is not robust enough. For the document grounding task, you can use the following settings: `min_pixels=1000 * 28 * 28` and `max_pixels=2000 * 28 * 28`.

**Comment by puppyapple** [comment]
> Currently 2.5VL is not robust enough. For the document grounding task, you can use the following settings: `min_pixels=1000 * 28 * 28` and `max_pixels=2000 * 28 * 28`.

@JJJYmmm Thanks for the quick reply, I will give it a try.

**Comment by puppyapple** [comment]
@JJJYmmm 我尝试了这个配置之后，用出现我上述情况的case试了下，有的没问题了，有的还是出现完全错误的情况，然后逐步调小预测时候的max_pixels时候，效果依然能逐渐变好。
有点好奇这个现象是如何发生的呀？
> 一开始我第一反应以为是模型有点过拟合在背答案，然后改动max_pixels之后导致了input_width和input_height的变动之后image tokens就不一样了，反而能正常预测。

**Comment by whycantfindaname** [comment]
> [@JJJYmmm](https://github.com/JJJYmmm) 我尝试了这个配置之后，用出现我上述情况的case试了下，有的没问题了，有的还是出现完全错误的情况，然后逐步调小预测时候的max_pixels时候，效果依然能逐渐变好。 有点好奇这个现象是如何发生的呀？
> 
> > 一开始我第一反应以为是模型有点过拟合在背答案，然后改动max_pixels之后导致了input_width和input_height的变动之后image tokens就不一样了，反而能正常预测。

大佬好，想问一下这个现象是您用max_pixels=2048 * 28 * 28训练好的模型在推理时调整 max_pixels后出现的吗，还是在训练时把 max_pixels就调整成 2000 * 28 * 28 了呢？如果方便的话有机会加个 vx 一起讨论一下吗

**Comment by Moyhub** [comment]
@JJJYmmm 是微调时候就应该将max_pixels设置的较小一些么？还是训练时候还是2048 * 28 * 28 推理时候适当变小？

**Comment by joaomsimoes** [comment]
Just a quick help from my experience. I was also struggling a lot to make it work with document layout and to have the right bounding boxes.

For my use case I did the following.

I have a huge handwritten dataset that is labeled with line by line bounding boxes. I have resized all images using `smart_resize` and use it for model training.

First I tried with SFT training and I was still struggling to have a good IoU. I had around 65%. 

After I tried to use GRPO and IoU as a reward and it really improved the results. I got to 77% IoU with just 1/3 of the training data.

From my understanding, the model can really learn better how to position the bounding boxes. I have fixed all images to have the same resolution (calculated by `smart_resize`)

I have not changed the settings from max_pixels and min_pixels. I'm using vLLM for inference and I wanted to use the default settings from the original model.

Just to summarize, I think it is possible to improve a lot the bounding boxes by using some RL technique. GRPO really works well but it needs a good dataset and it is expensive to train.
